src/YAxis.re

```diff
@@ -37,20 +41,73 @@ external make:
     ~padding: paddingVertical=?,
     ~reversed: bool=?,
     ~scale: scale=?,
+    ~stroke: string=?,
     ~style: ReactDOM.Style.t=?,
     ~tick: 'tick=?,
     ~tickCount: int=?,
     ~tickFormatter: 'tickFormatter=?,
     ~tickLine: 'tickLine=?,
     ~tickMargin: int=?,
     ~ticks: array('ticks)=?,
-    ~tickSize: int=?,
+    ~tickSize: float=?,
+    ~transform: string=?,
     ~unit: string=?,
     ~width: int=?,
     ~yAxisId: string=?
   ) =>
   React.element =
   "YAxis";
 
-let makeProps = (~interval=?) =>
-  makeProps(~interval=?interval->AxisInterval.encodeOpt);
+let makeProps =
+    (
+      ~interval=?,
+      ~onClick=?,
+      ~onMouseDown=?,
+      ~onMouseEnter=?,
+      ~onMouseLeave=?,
+      ~onMouseMove=?,
+      ~onMouseOut=?,
+      ~onMouseOver=?,
+      ~onMouseUp=?,
+    ) =>
+  makeProps(
+    ~interval=?interval |> AxisInterval.encodeOpt,
+    ~onClick=?
```

Comment:
This adds some extra runtime ([example](https://melange.re/unstable/playground/?language=Reason&code=W0BtZWwubW9kdWxlICJyZWNoYXJ0cyJdIFtAcmVhY3QuY29tcG9uZW50XQpleHRlcm5hbCBtYWtlOgogICgKICAgIH5hbGxvd0RhdGFPdmVyZmxvdzogYm9vbD0%2FLAogICAgfmFsbG93RGVjaW1hbHM6IGJvb2w9PywKICAgIH5hbGxvd0R1cGxpY2F0ZWRDYXRlZ29yeTogYm9vbD0%2FLAogICAgfmF4aXNMaW5lOiAnYXhpc0xpbmU9PywKICAgIH5jbGFzc05hbWU6IHN0cmluZz0%2FLAogICAgfmRhdGFLZXk6ICdkYXRhS2V5PT8sCiAgICB%2BZG9tYWluOiBhcnJheSgnZG9tYWluKT0%2FLAogICAgfmhlaWdodDogaW50PT8sCiAgICB%2BaGlkZTogYm9vbD0%2FLAogICAgfmxhYmVsOiAnbGFiZWw9PywKICAgIH5taW5UaWNrR2FwOiBpbnQ9PywKICAgIH5taXJyb3I6IGJvb2w9PywKICAgIH5uYW1lOiBzdHJpbmc9PywKICAgIH5vbkNsaWNrOiAoLiBKcy5OdWxsYWJsZS50KEpzLnQoey4ufSkpLCBSZWFjdC5FdmVudC5Nb3VzZS50KSA9PiB1bml0PT8sCiAgICB%2Bb25Nb3VzZURvd246CiAgICAgICguIEpzLk51bGxhYmxlLnQoSnMudCh7Li59KSksIFJlYWN0LkV2ZW50Lk1vdXNlLnQpID0%2BIHVuaXQ9PywKICAgIH5vbk1vdXNlRW50ZXI6CiAgICAgICguIEpzLk51bGxhYmxlLnQoSnMudCh7Li59KSksIFJlYWN0LkV2ZW50Lk1vdXNlLnQpID0%2BIHVuaXQ9PywKICAgIH5vbk1vdXNlTGVhdmU6ICguIEpzLnQoey4ufSksIFJlYWN0LkV2ZW50Lk1vdXNlLnQpID0%2BIHVuaXQ9PywKICAgIH5vbk1vdXNlTW92ZToKICAgICAgKC4gSnMuTnVsbGFibGUudChKcy50KHsuLn0pKSwgUmVhY3QuRXZlbnQuTW91c2UudCkgPT4gdW5pdD0%2FLAogICAgfm9uTW91c2VPdXQ6CiAgICAgICguIEpzLk51bGxhYmxlLnQoSnMudCh7Li59KSksIFJlYWN0LkV2ZW50Lk1vdXNlLnQpID0%2BIHVuaXQ9PywKICAgIH5vbk1vdXNlT3ZlcjoKICAgICAgKC4gSnMuTnVsbGFibGUudChKcy50KHsuLn0pKSwgUmVhY3QuRXZlbnQuTW91c2UudCkgPT4gdW5pdD0%2FLAogICAgfm9uTW91c2VVcDogKC4gSnMuTnVsbGFibGUudChKcy50KHsuLn0pKSwgUmVhY3QuRXZlbnQuTW91c2UudCkgPT4gdW5pdD0%2FLAogICAgfm9yaWVudGF0aW9uOgogICAgICBbCiAgICAgICAgfCBgbGVmdAogICAgICAgIHwgYHJpZ2h0CiAgICAgIF0KICAgICAgICA9PywKICAgIH5yZXZlcnNlZDogYm9vbD0%2FLAogICAgfnN0cm9rZTogc3RyaW5nPT8sCiAgICB%2Bc3R5bGU6IFJlYWN0RE9NLlN0eWxlLnQ9PywKICAgIH50aWNrOiAndGljaz0%2FLAogICAgfnRpY2tDb3VudDogaW50PT8sCiAgICB%2BdGlja0Zvcm1hdHRlcjogJ3RpY2tGb3JtYXR0ZXI9PywKICAgIH50aWNrTGluZTogJ3RpY2tMaW5lPT8sCiAgICB%2BdGlja01hcmdpbjogaW50PT8sCiAgICB%2BdGlja3M6IGFycmF5KCd0aWNrcyk9PywKICAgIH50aWNrU2l6ZTogZmxvYXQ9PywKICAgIH50cmFuc2Zvcm06IHN0cmluZz0%2FLAogICAgfnVuaXQ6IHN0cmluZz0%2FLAogICAgfndpZHRoOiBpbnQ9PywKICAgIH55QXhpc0lkOiBzdHJpbmc9PwogICkgPT4KICBSZWFjdC5lbGVtZW50ID0KICAiWUF4aXMiOwoKbGV0IG1ha2VQcm9wcyA9CiAgICAoCiAgICAgIH5pbnRlcnZhbD0%2FLAogICAgICB%2Bb25DbGljaz0%2FLAogICAgICB%2Bb25Nb3VzZURvd249PywKICAgICAgfm9uTW91c2VFbnRlcj0%2FLAogICAgICB%2Bb25Nb3VzZUxlYXZlPT8sCiAgICAgIH5vbk1vdXNlTW92ZT0%2FLAogICAgICB%2Bb25Nb3VzZU91dD0%2FLAogICAgICB%2Bb25Nb3VzZU92ZXI9PywKICAgICAgfm9uTW91c2VVcD0%2FLAogICAgKSA9PgogIG1ha2VQcm9wcygKICAgIH5vbkNsaWNrPT8KICAgICAgb25DbGljawogICAgICB8PiBPcHRpb24ubWFwKG9uQ2xpY2sgPT4gKC4gcGF5bG9hZCwgZXZlbnQpID0%2BIG9uQ2xpY2socGF5bG9hZCwgZXZlbnQpKSwKICAgIH5vbk1vdXNlRG93bj0%2FCiAgICAgIG9uTW91c2VEb3duCiAgICAgIHw%2BIE9wdGlvbi5tYXAob25Nb3VzZURvd24gPT4KICAgICAgICAgICAoLiBwYXlsb2FkLCBldmVudCkgPT4gb25Nb3VzZURvd24ocGF5bG9hZCwgZXZlbnQpCiAgICAgICAgICksCiAgICB%2Bb25Nb3VzZUVudGVyPT8KICAgICAgb25Nb3VzZUVudGVyCiAgICAgIHw%2BIE9wdGlvbi5tYXAob25Nb3VzZUVudGVyID0%2BCiAgICAgICAgICAgKC4gcGF5bG9hZCwgZXZlbnQpID0%2BIG9uTW91c2VFbnRlcihwYXlsb2FkLCBldmVudCkKICAgICAgICAgKSwKICAgIH5vbk1vdXNlTGVhdmU9PwogICAgICBvbk1vdXNlTGVhdmUKICAgICAgfD4gT3B0aW9uLm1hcChvbk1vdXNlTGVhdmUgPT4KICAgICAgICAgICAoLiBwYXlsb2FkLCBldmVudCkgPT4gb25Nb3VzZUxlYXZlKHBheWxvYWQsIGV2ZW50KQogICAgICAgICApLAogICAgfm9uTW91c2VNb3ZlPT8KICAgICAgb25Nb3VzZU1vdmUKICAgICAgfD4gT3B0aW9uLm1hcChvbk1vdXNlTW92ZSA9PgogICAgICAgICAgICguIHBheWxvYWQsIGV2ZW50KSA9PiBvbk1vdXNlTW92ZShwYXlsb2FkLCBldmVudCkKICAgICAgICAgKSwKICAgIH5vbk1vdXNlT3V0PT8KICAgICAgb25Nb3VzZU91dAogICAgICB8PiBPcHRpb24ubWFwKG9uTW91c2VPdXQgPT4KICAgICAgICAgICAoLiBwYXlsb2FkLCBldmVudCkgPT4gb25Nb3VzZU91dChwYXlsb2FkLCBldmVudCkKICAgICAgICAgKSwKICAgIH5vbk1vdXNlT3Zlcj0%2FCiAgICAgIG9uTW91c2VPdmVyCiAgICAgIHw%2BIE9wdGlvbi5tYXAob25Nb3VzZU92ZXIgPT4KICAgICAgICAgICAoLiBwYXlsb2FkLCBldmVudCkgPT4gb25Nb3VzZU92ZXIocGF5bG9hZCwgZXZlbnQpCiAgICAgICAgICksCiAgICB%2Bb25Nb3VzZVVwPT8KICAgICAgb25Nb3VzZVVwCiAgICAgIHw%2BIE9wdGlvbi5tYXAob25Nb3VzZVVwID0%2BCiAgICAgICAgICAgKC4gcGF5bG9hZCwgZXZlbnQpID0%2BIG9uTW91c2VVcChwYXlsb2FkLCBldmVudCkKICAgICAgICAgKSwKICApOw%3D%3D&live=off)). But I guess we're fine with it? cc @davesnx

Reply:
It already had the runtime on it with `interval |> AxisInterval.encodeOpt` [here](https://github.com/ahrefs/melange-recharts/blob/master/src/XAxis.re#L55-L56) so I figured it was ok but I can rip these out.

I included them mostly to help upgrading since you'd have to rewrite all of the functions to be uncurried otherwise but since these have breaking changes in them anyways maybe that's ok?